h5py/h5d.pyx

```diff
@@ -430,3 +436,64 @@ cdef class DatasetID(ObjectID):
                 efree(offset)
                 if space_id:
                     H5Sclose(space_id)
+
+    IF HDF5_VERSION >= (1, 10, 2):
+
+        def read_direct_chunk(self, offsets, filter_mask=0x0000, PropID dxpl=None):
+            """ (offsets, uint32_t filter_mask=0x0000, PropID dxpl=None)
+
+            Reads data to a bytes array directly from a chunk at position
+            specified by the offsets argument.
+
+            The `filter_mask` is a bit field of up to 32 values. It specify
+            which filters in the pipeline are used with the chunk.
+            Using the default `0x0000` with use all the filters, while `0xFFFF`
+            will skip all the filters.
```

Comment:
I've just gone and looked at the HDF5 code, and I think `filter_mask` is actually not an input parameter for reading the chunk at all - it's just a pointer to receive the information HDF5 has recorded about filters applied to that chunk when writing. Sorry, I should have tried harder to understand this before.

Reply:
I don't think so cause it is an in/out argument. That's maybe a mistake on the documentation.
It needs more investigation to check with a real dataset.

Reply:
I think the docs are mistaken, or at least confusing. But the value it points to appears to be [zeroed out](https://bitbucket.hdfgroup.org/projects/HDFFV/repos/hdf5/browse/src/H5Dchunk.c?at=refs%2Ftags%2Fhdf5-1_10_5#524) before being [set to a retrieved value](https://bitbucket.hdfgroup.org/projects/HDFFV/repos/hdf5/browse/src/H5Dchunk.c?at=refs%2Ftags%2Fhdf5-1_10_5#582-583). I can't see anywhere that the value is used as input.

This also fits with the docs saying that the 'direct' read and write interfaces bypass the filter pipeline entirely.

Reply:
Thanks. It makes more sense now. I also have to rewrite the function to remove the input filter mask.